src/lib.rs

```diff
@@ -1145,6 +1147,58 @@ impl<'a> fmt::Display for Urn<'a> {
     }
 }
 
+#[cfg(feature = "uuid_macro")]
+#[macro_use]
+extern crate proc_macro_hack;
+
+#[cfg(feature = "uuid_macro")]
+#[macro_use]
+#[allow(unused_imports)]
+extern crate uuid_macro;
+
+#[cfg(feature = "uuid_macro")]
+#[doc(hidden)]
+pub use uuid_macro::*;
+
+#[cfg(feature = "uuid_macro")]
+proc_macro_expr_decl! {
+    #[doc(hidden)]
+    uuid_parts! => uuid_parts_impl
+}
+
+#[cfg(all(feature = "uuid_macro",feature = "nightly"))]
+pub const fn new(bytes:[u8;16]) -> Uuid {
```

Comment:
instead of this set of `new` functions... use `Uuid::from_uuid_bytes`.. alter them as need for `const`ness